dubbo-common/src/main/java/org/apache/dubbo/common/logger/LoggerFactory.java

```diff
@@ -103,12 +108,7 @@ public static void setLoggerAdapter(LoggerAdapter loggerAdapter) {
      * @return logger
      */
     public static Logger getLogger(Class<?> key) {
-        FailsafeLogger logger = LOGGERS.get(key.getName());
-        if (logger == null) {
-            LOGGERS.putIfAbsent(key.getName(), new FailsafeLogger(LOGGER_ADAPTER.getLogger(key)));
-            logger = LOGGERS.get(key.getName());
-        }
-        return logger;
+        return LOGGERS.computeIfAbsent(key.getName(), name -> new FailsafeLogger(LOGGER_ADAPTER.getLogger(key)));
```

Comment:
I would suggest :

```java
return LOGGERS.computeIfAbsent(key.getName(), name -> new FailsafeLogger(LOGGER_ADAPTER.getLogger(name)));
```
for a better readability.
Note parameter for `getLogger` has been changed from `key` to `name`.

Reply:
@ralf0131 If that's the case, why not just call the overloaded version `public static Logger getLogger(String key)`

Reply:
When you are writing lambda x -> f(x) , `x` is just a placeholder, but in the function f(x) you treat  x as input parameters. 

Equivalently, in your case, it should be `name -> new FailsafeLogger(LOGGER_ADAPTER.getLogger(name))`, and `key` will be passed to the function at runtime.

You've implemented the same logic at Line 121.

When you are writing `name -> new FailsafeLogger(LOGGER_ADAPTER.getLogger(key))`, you are treating the function as zero parameter function, where `key` is a constant in this case.

Reply:
I know what you meant, I'll update this patch to show what I meant

Reply:
@ralf0131 Updated.

Reply:
No, I didn't mean to call the overloaded method. Instead, I prefer your initial implementation. Call the overloaded method will introduce more overhead.

Your initial implementation is logically correct, what I mean is a minor change to improve the readability. You just need to replace `key` to `name`.

Reply:
> You just need to replace key to name
That will also change the logic before this patch, it assumes that `org.apache.dubbo.common.logger.LoggerAdapter#getLogger(java.lang.Class<?>)` and `org.apache.dubbo.common.logger.LoggerAdapter#getLogger(java.lang.String)` return the same logger(which may be true in this case).

My thought was that `org.apache.dubbo.common.logger.LoggerFactory#getLogger(java.lang.Class<?>)` will get the logger via the key which is type of `Class` and, if I "replace key with name", the actual logger is got via name(which is type of `String`), anyway, it's the same in this situation, I'll update it soon

Reply:
@ralf0131 updated